.github/workflows/ci.yml

```diff
@@ -0,0 +1,123 @@
+name: CI
+on:
+  # Run on all pushes to master (including merge of PR)
+  push:
+    branches: master
+  # Run on all changes in pull requests
+  pull_request:
+  # Run CI once per day (at 06:00 UTC)
+  # schedule:
+  #   - cron: '0 6 * * *'
+
+jobs:
+  nikola:
+    name: Basic tests (Python ${{ matrix.python }} on ${{ matrix.image }})
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - 3.5
+          - 3.6
+          - 3.7
+          - 3.8
+        image:
+          - ubuntu-18.04
+          - ubuntu-16.04
+        exclude:
+          - python: 3.5
+            image: ubuntu-18.04
+          - python: 3.6
+            image: ubuntu-16.04
+          - python: 3.7
+            image: ubuntu-16.04
+          - python: 3.8
+            image: ubuntu-16.04
+    runs-on: '${{ matrix.image }}'
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '${{ matrix.python }}'
+      - name: Install test requirements
+        run: |
+          pip install --upgrade-strategy eager -Ur requirements-tests.txt
+      - name: Install coverage requirements
+        run: |
+          pip install --upgrade-strategy eager -U codacy-coverage
+      - name: Install Nikola
+        run: |
+          pip install .
+      - name: Run tests
+        run: |
+          py.test tests/
+      - name: Run nikola
+        run: |
+          nikola
+      - name: Run nikola help
+        run: |
+          nikola help
+      - name: Compare to baseline
+        run: |
+          scripts/baseline.sh check
+      - name: Code coverage
+        run: |
+          scripts/codacy_coverage.sh
+
+  flake8:
+    name: Linting
+    strategy:
+      matrix:
+        python:
+          - 3.8
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '${{ matrix.python }}'
+      - name: Install requirements
+        run: |
+          pip install flake8 pydocstyle
+      - name: Run flake8
+        run: |
+          flake8 nikola/ tests/
+      - name: Run pydocstyle
+        run: |
+          pydocstyle --count --match-dir='(?!^\\.)(?!data).*' nikola/
+
+  basereq:
+    name: Install with base requirements
+    strategy:
+      matrix:
+        python:
+          - 3.8
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '${{ matrix.python }}'
+      - name: Install requirements
```

Comment:
This also requires `pip install -U --upgrade-strategy=eager pip setuptools wheel`